Update all dependencies to latest version

murattuzel opened the pull request:
Updated dependencies and made necessary changes.

murattuzel committed:
Update all dependencies to latest version

Update deprecated dagger injector and builder

Update deprecated okhttp fields

Update deprecated view model providers

Add workaround for ktlint import ordering issue

murattuzel committed:
Remove dependencyUpdates task from pre-commit git hook

Fix reformat task line link

fatihergin commented on app/src/main/kotlin/com/adesso/movee/internal/injection/module/NetworkModule.kt:
```diff
@@ -101,7 +101,7 @@ internal class NetworkModule {
             .baseUrl(BuildConfig.BASE_URL)
             .addConverterFactory(MoshiConverterFactory.create(moshi))
-            .callFactory { client.get().newCall(it) }
+            .callFactory(client.get())
```
no need to use deferred initialization of okhttp client anymore.
consider removing it from method constructor and deependency list if exist

fatihergin commented on config/dependencies.gradle:
```diff
@@ -2,13 +2,13 @@ ext {
             minSdk    : 21,
-            targetSdk : 28,
-            compileSdk: 28
+            targetSdk : 30,
+            compileSdk: 30
```
we should be aware of compatibility issues. 
is there any deprecated usages or behavioral changes between sdk 28 and 30 like deprecation of network info class?

murattuzel replied:
I added new connectivity checking logic. Prior to sdk 23 can be continue using old logic but new logic should be used from sdk 23.

fatihergin commented on config/dependencies.gradle:
```diff
@@ -17,41 +17,42 @@ ext {
-            lifecycle       : '1.1.0',
-            androidKtx      : '0.1',
-            coroutines      : '1.3.2',
+            dagger          : '2.29.1',
```
there was a compatibility issue of dagger v2.23+ with our codebase. 
we must be sure about creating and providing all base views(activity & fragment) seamlessly.
Testing a BottomSheetDialogFragment instance would be enough to be sure about this issue.

murattuzel replied:
Tested locally and seems fine. [Proof](https://user-images.githubusercontent.com/17027155/97162955-fbabc580-1790-11eb-9ff4-d3a16944d3c2.png)

fatihergin commented on config/gradle-versions-plugin.gradle:
```diff
@@ -0,0 +1,33 @@
+tasks.named("dependencyUpdates").configure {
+    checkForGradleUpdate = true
+
+    // Example 1: reject all non stable versions
```
method and variable names are pretty clear.
the comment lines can be removed.

fatihergin commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-bin.zip
```
gradle distributions packaged with --all should be preferred instead of bin.
check [this](https://docs.gradle.org/current/userguide/gradle_wrapper.html) for more info about gradle wrapper

murattuzel committed:
Fix deprecated network connectivity check

Fix lazy initialization of okHttpClient

Change gradle wrapper -bin distribution with -all

Remove unnecessary comments from gradle versions plugin config
